app/src/main/java/ch/admin/bag/dp3t/whattodo/WtdPositiveTestFragment.java

```diff
@@ -47,10 +55,70 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 			UrlUtil.openUrl(getContext(), getString(R.string.faq_button_url));
 		});
 
-		view.findViewById(R.id.wtd_inform_call_infoline_coronavirus).setOnClickListener(v->{
-			PhoneUtil.callInfolineCoronavirus(v.getContext());
-		});
+		View oldCallButton = view.findViewById(R.id.wtd_inform_call_infoline_coronavirus);
+		if (oldCallButton != null) {
+			oldCallButton.setOnClickListener(v -> {
+				PhoneUtil.callInfolineCoronavirus(v.getContext());
+			});
+		}
+	}
+
+	private void fillContentFromConfigServer(View view) {
+		Context context = view.getContext();
+		SecureStorage secureStorage = SecureStorage.getInstance(context);
+		WhatToDoPositiveTestTextsModel textModel =
+				secureStorage.getWhatToDoPositiveTestTexts(context.getString(R.string.language_key));
+
+		if (textModel != null) {
+			((TextView) view.findViewById(R.id.wtd_inform_box_supertitle)).setText(textModel.getEnterCovidcodeBoxSupertitle());
+			((TextView) view.findViewById(R.id.wtd_inform_box_title)).setText(textModel.getEnterCovidcodeBoxButtonTitle());
+			((TextView) view.findViewById(R.id.wtd_inform_box_text)).setText(textModel.getEnterCovidcodeBoxText());
+			((TextView) view.findViewById(R.id.wtd_inform_button)).setText(textModel.getEnterCovidcodeBoxButtonTitle());
+
+			if (textModel.getInfoBox() != null) {
+				((TextView) view.findViewById(R.id.wtd_inform_infobox_title)).setText(textModel.getInfoBox().getTitle());
+				((TextView) view.findViewById(R.id.wtd_inform_infobox_msg)).setText(textModel.getInfoBox().getMsg());
+
+				if (textModel.getInfoBox().getUrl() != null && textModel.getInfoBox().getUrlTitle() != null) {
+					((TextView) view.findViewById(R.id.wtd_inform_infobox_link_text)).setText(textModel.getInfoBox().getUrlTitle());
+					view.findViewById(R.id.wtd_inform_infobox_link_layout).setOnClickListener(v -> {
+						UrlUtil.openUrl(v.getContext(), textModel.getInfoBox().getUrl());
+					});
+					view.findViewById(R.id.wtd_inform_infobox).setVisibility(View.VISIBLE);
+				} else {
```

Comment:
The visibility should be set on the level of `if (textModel.getInfoBox() != null)` (one if/else level up), not here.

Reply:
fixed